### DISCUSSION TITLE
Best way to generate a single page PDF using selected items from current page in MkDocs

### DISCUSSION THREAD

**Discussion by andrewchung** [mkdocs/mkdocs#3861] [Q&A]
Let's say i have a page in MkDocs that has a several paragraphs of text together with a checklist.

I now want the user to be able to click a button, and generate a PDF with just the checklist on it. I do not want the paragraphs of text to print, and I want to be able to control how the checklist looks on paper vs the screen (e.g. change font font size).

Is there a tool or plug-in that would be suitable for this type of use case?

Thank you

**Comment by joapuiib** [comment]
Hello.
You can generate a PDF by printing the web page.

Then, you can use a [custom CSS](https://www.mkdocs.org/user-guide/customizing-your-theme/) and select which elements are shown with the media query `@media print`.

In this example, I've removed the block with the class `.md-source-file` from beign printed.
```css
@media print {
    .md-source-file {
        display: none;
    }
}
```

Maybe you can create a custom CSS class to include/exclude elements when printing.
Classes can be set to elements in Markdown using the [Attribute Lists Extension](https://python-markdown.github.io/extensions/attr_list/)

If you want to build the PDF directly, maybe plugins such as [mkdocs-pdf-export-plugin](https://github.com/zhaoterryy/mkdocs-pdf-export-plugin) could be of help.
